src/rebar_state.erl

```diff
@@ -492,6 +494,11 @@ add_provider(State=#state_t{providers=Providers, allow_provider_overrides=false}
             State#state_t{providers=[Provider | Providers]}
     end.
 
+-spec maybe_default_cdn(t()) -> binary().
+maybe_default_cdn(State) ->
+    CDN = rebar_state:get(State, rebar_packages_cdn, ?DEFAULT_CDN),
+    rebar_utils:to_binary(rebar_utils:to_binary(CDN)).
+
```

Comment:
A few problems here:

- In this context this function should be called `default_hex_repo_url` or something like that, since the concept of a CDN for it is no longer really accurate
- there is a double-call to `rebar_utils:to_binary` that can be a single one.